src/Type/Php/FilterFunctionReturnTypeHelper.php

```diff
@@ -240,7 +242,7 @@ private function getConstant(string $constantName): int
 		$constant = $this->reflectionProvider->getConstant(new Node\Name($constantName), null);
 		$valueType = $constant->getValueType();
 		if (!$valueType instanceof ConstantIntegerType) {
-			throw new ShouldNotHappenException(sprintf('Constant %s does not have integer type.', $constantName));
+			return self::UNKNOWN_CONSTANT;
```

Comment:
I thought about returning null, or catching the exception at the call-sites, but this seemed too much for this case. instead I decided to return a negative constant value, which does not overlap with other existing FILTER_* constants